Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -256,21 +264,27 @@ public Task InvokeDidChangeConfigurationAsync(LSP.DidChangeConfigurationParams r
             await SendDocumentOpensAsync();
         }
 
-        public Task<LSP.CompletionList> InvokeTextDocumentCompletionAsync(LSP.CompletionParams request, CancellationToken cancellationToken = default)
-            => _rpc == null ? Task.FromResult(new LSP.CompletionList()) : _rpc.InvokeWithParameterObjectAsync<LSP.CompletionList>("textDocument/completion", request, cancellationToken);
+        public Task<object> InvokeTextDocumentCompletionAsync(LSP.CompletionParams request, CancellationToken cancellationToken = default)
+            => _rpc == null ? Task.FromResult<object>(null) : _rpc.InvokeWithParameterObjectAsync<object>("textDocument/completion", request, cancellationToken);
```

Comment:
how come we dont know the result type anymore?

Reply:
The result type based on the spec is not LSP.CompletionList. It can be one of LSP.CompletionList or LSP.CompletionItem[] or null.

Reply:
Exact text of the spec:

```
Response:

result: CompletionItem[] | CompletionList | null. If a CompletionItem[] is provided it is interpreted to be complete. So it is the same as { isIncomplete: false, items }
```

Reply:
And since C# doesn't support type unions, it has to be an object.